### PR TITLE
chore: added a CI step to detect unused dependencies

### PR DESCRIPTION
**Pull request by divinetettey**
adding this PR to resolve this issue  [Implement a CI step to detect unused dependencies 📦](https://github.com/finos/git-proxy/issues/413)

The changes include: 
1. installing the [depcheck](https://www.npmjs.com/package/depcheck) package 
2. Running `depcheck --oneline --skip-missing`
3.  Fail if the the check fails

### CODE REVIEW

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -18,4 +18,20 @@ jobs:
...
         fail-on-scopes: development, runtime
-        allow-dependencies-licenses: 'pkg:npm/caniuse-lite'
\ No newline at end of file
+        allow-dependencies-licenses: 'pkg:npm/caniuse-lite'
+     - name: 'Setup Node.js'
+       uses: actions/setup-node@v4
+       with:
+        node-version: '18.x'
+     - name: 'Install dependencies'
+       run: npm install
```
Comment (JamieSlome) [inline .github/workflows/dependency-review.yml:27]: Do we need this or the `npm install` command if we are invoking `depcheck` via `npx`? 🤔

Comment (divinecodes) [reply .github/workflows/dependency-review.yml:27]: yeah, I don't think we need. Thanks for catching that. Pushing an update

Comment (divinetettey) [reply .github/workflows/dependency-review.yml:27]: yh, I don't think we need that. Thanks for catching it. Pushing an update

File: .github/workflows/dependency-review.yml [hunk 2]
```diff
@@ -18,4 +18,18 @@ jobs:
...
         fail-on-scopes: development, runtime
-        allow-dependencies-licenses: 'pkg:npm/caniuse-lite'
\ No newline at end of file
+        allow-dependencies-licenses: 'pkg:npm/caniuse-lite'
+     - name: 'Setup Node.js'
+       uses: actions/setup-node@v4
+       with:
+        node-version: '18.x'
+     - name: 'Install depcheck globally'
+       run: npm install -g depcheck
```
Comment (JamieSlome) [inline .github/workflows/dependency-review.yml:27]:
```suggestion
```

**Comment by JamieSlome** [comment]
@divinetettey - is there any chance we can move the check to its own workflow file? What do you think?

**Comment by divinetettey** [comment]
> @divinetettey - is there any chance we can move the check to its own workflow file? What do you think?

I think that's fine, we can go with approach

**Comment by JamieSlome** [comment]
@divinetettey - thanks for making the adjustments (perfect 🎉 ). It seems like the `depcheck` is reporting that `concurrently` and a few other `devDependencies` are unused which is not the case. Is there any way to add these to an allow list or as permissible exceptions?

**Comment by divinetettey** [comment]
> @divinetettey - thanks for making the adjustments (perfect 🎉 ). It seems like the `depcheck` is reporting that `concurrently` and a few other `devDependencies` are unused which is not the case. Is there any way to add these to an allow list or as permissible exceptions?

I have added them to the ignore list

File: .github/workflows/unused-dependencies.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: 'Unused Dependencies Check'
```
Comment (JamieSlome) [inline .github/workflows/unused-dependencies.yml:1]:
```suggestion
name: 'Unused Dependencies'
```

File: .github/workflows/unused-dependencies.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: 'Unused Dependencies'
+on: [pull_request]
+
+permissions: 
+  contents: read 
+  pull-requests: write
```
Comment (JamieSlome) [inline .github/workflows/unused-dependencies.yml:6]:
```suggestion
```
Are `write` permissions actually required here?

**Comment by JamieSlome** [comment]
@maoo - can we make unused dependencies required on status checks to `main` please?

**Review by JamieSlome — APPROVED** [review]
LGTM! 🍰

**Comment by maoo** [comment]
> @maoo - can we make unused dependencies required on status checks to `main` please?

Done!

**Pull request merged**